e3/anod/checkout.py

```diff
@@ -161,6 +181,20 @@ def update_git(self, url, revision):
             if old_commit == new_commit:
                 result = ReturnValue.unchanged
             else:
+                # Fetch the change log and dump it into the changelog file
+                with open(self.changelog_file, 'w') as fd:
```

Comment:
That's strange to write some contents in a file and then rewrite it. I would use a temp file for that first dump to make the code easier to read.